services/src/main/java/org/keycloak/services/resources/admin/GroupResource.java

```diff
@@ -121,6 +123,8 @@ public Response updateGroup(GroupRepresentation rep) {
             throw ErrorResponse.error("Group name is missing", Response.Status.BAD_REQUEST);
         }
 
+        checkForOrgRelatedGroupModelAndRep(session, group, rep);
```

Comment:
You should be able to avoid all the checks in this class by a single one at https://github.com/keycloak/keycloak/blob/main/services/src/main/java/org/keycloak/services/resources/admin/GroupsResource.java#L128.

Reply:
If I add this check one level above I would block all other operation (like get, getMembers, ...) as well, which is not desired.

Reply:
Yeah, and that is the goal. It does not make sense to edit a group associated with an org. Does it?

Reply:
get, getMembers are edit operations? I would say those are read-only operations and I'm allowing those because we agreed, that we will block only update and delete operations, not read (or list) operations.

Reply:
Actually, we change the scope of this task a bit. The initial idea was to "do not return groups associated with organizations" (as per issue/PR description) but then we decided to not change queries now as they are going to touch an already problematic area for which we have an entire epic to improve its behavior and scalability.